.github/actions/build_pandas/action.yml

```diff
@@ -16,7 +16,4 @@ runs:
         python -m pip install -e . --no-build-isolation --no-use-pep517 --no-index
       shell: bash -el {0}
       env:
-        # Cannot use parallel compilation on Windows, see https://github.com/pandas-dev/pandas/issues/30873
```

Comment:
Unless this address's @simonjayhawkins https://github.com/pandas-dev/pandas/pull/47081#issuecomment-1157504061= I would prefer to keep this at `N_JOBS: 1`

Otherwise, I would suggest this be specific to the number of cores available on GHA hosted runners by OS: https://github.com/pandas-dev/pandas/pull/47081#discussion_r878982485=

Reply:
> Unless this address's @simonjayhawkins [#47081 (comment)](https://github.com/pandas-dev/pandas/pull/47081#issuecomment-1157504061)

I've needed to restart a few workflows this last week or so as they fail on the initial build, but it maybe once it is fixed that the initial build is successful, that subsequent incremental builds for commits in the range of the last couple of weeks could also be successful.

And it could also be that if this is a reoccurring problem that I can change the workflow to repeat the previous build step if the pandas import fails.

so I am happy that any comments/concerns are fully addressed before merging this.